contributors/design-proposals/standardization-of-development-process.md

```diff
@@ -0,0 +1,282 @@
+```
+
+## Responsible SIG(s)
+
+- SIG Release
+- SIG PM
+- SIG Architecture
+- SIG Governance
+- Technical Steering Committee
+
+## Summary
+
+A standardized development process for Kubernetes is proposed in order to:
+- provide a common structure for proposing changes to Kubernetes
+- ensure that the motivation for a change is clear
+- allow for the enumeration stability milestones and stability graduation
+  criteria
+- persist project information in a Version Control System (VCS) for future
+  Kubernauts
+- support the creation of _high value user facing_ information such as:
+  - release notes
+  - release announcement blog
+  - a cross project development road map
+- support development across multiple repositories beyond `kubernetes/kubernetes`
+- begin to move away from using GitHub issues for feature requests and reserve them
+  for tracking work in flight
+
+## Motivation
+
+Today it is becoming fairly difficult to track changes being made across all
+Special Interest Groups (SIGs) within Kubernetes. For cross project SIGs such
+as SIG PM and SIG Release an abstraction beyond a single GitHub Issue or Pull
+request seems to be required in order to understand and communicate upcoming
+changes to Kubernetes. Particularly the generation of release notes and the
+release announcement blog post are rather difficult and have sometimes delayed
+a release due to incompleteness. In a blog post describing the [road to Go 2][],
+Russ Cox explains
+
+> that it is difficult but essential to describe the significance of a problem
+> in a way that someone working in a different environment can understand
+
+unfortunately I believe that our current mechanism for proposing changes by
+opening a GitHub issue against `kubernetes/features` falls short of providing
+an effective vehicle for agreeing on the motivation and the design of a proposed
+change.
+
+The use of GitHub issues when proposing changes does not provide SIGs good
+facilities for signaling approval or rejection of a proposed change to Kubernetes
+since anyone can open a GitHub issue at any time. Additionally managing a proposed
+change across multiple releases is somewhat cumbersome as labels and milestones
+need to be updated for every release that a change spans which leads to an ever
+increasing number of issues open against `kubernetes/features` which itself has
+become a management problem.
+
+We have chosen to use Git as our VCS with GitHub currently used to host our
+repositories. While GitHub provides a rich feature set as an open source project,
+managing issues over time and searching for text within an issue can be
+challenging. The flat hierarchy of issues can also make navigation and
+categorization tricky. While not all community members might not be comfortable
+using Git directly, I believe it is imperative that as a community we work to
+educate people on a standard set of tools so they can take their experience to
+other projects they may decide to work on in the future.
+
```

Comment:
I agree on the ubiquity of Git.  That said, a VCS is not a project management tool, nor a cogent way of managing an architectural catalog/backlog.  An insistence on finding ways to "make Git work" for various processes is going to lead to suboptimal results.  We need to be explicit about its role in the ecosystem, as well as identify critical tooling gaps from the program management perspective.